lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).as_json
```

Comment:
Sorry, I lost context of the change we were talking about.  

```diff
class Serializer
+    def as_json
+      self.class.adapter.new(self).as_json
```

I agree it would be good to have a single simple way to serialize a resource, but I think this specific change, as simple as it is, sweeps the problem under the rug.  While, I myself was at first surprised that I couldn't call as_json on a Serializer, the fact is, as the diff highlights, we don't. We call it on the adapter.  So, adding an instance method to a serializer that creates an adapter and then calls as_json, is, I think, problematic because it introduces *more* places where the code is responsible for knowing to pass the serializer instance into a new adapter then call as_json on the resulting adapter instance.  As much as I really like the idea word-wise, as the code stands, I think it actually adds more complexity.

That said, if we merge something like #954 (disclaimer, I'm writing it), where we have a single place where serialization happens, then we could add this method as

```ruby
def as_json(json_options = {})
  ActiveModel::Serializer.build(object, options.merge(serializer: self)).as_json(json_options) # pseudo-code
def to_json
  ActiveModel::Serializer.build(object, options.merge(serializer: self)).to_json # pseudo-coe
```

Then we could have the convenience of BlogSerializer.new(blog).to_json and maintain the single place in the code of knowing how to combine the object, serializer, adapter, and options. (though it would still lose the ability to pass in any adapter options and still keep the as_json interface unless we start deleting keys... etc)